### PR TITLE
docker GCC and ALSA updates

### PR DESCRIPTION
**Pull request by lgirdwood**
Bump GCC to 9.3 and Keep ALSA source in the container to permit ALSA development.
@xiulipan I think I screwed up my container when testing this, I can build it manually but not with the build script (maybe I need to force clean it?) Can you check. Thanks.

### CODE REVIEW

**Comment by xiulipan** [comment]
@lgirdwood I am running the docker build process now. Will update you with result later

**Comment by xiulipan** [comment]
@lgirdwood Same build error to me. Let me try to build it locally and check what the issue is.

**Comment by xiulipan** [comment]
@lgirdwood It seems to be issue related to folder owner. The alsa can only be install by ROOT. But the compiler is build as sof user. The sof user will not have permission to folder

File: scripts/docker_build/sof_builder/Dockerfile [hunk 1]
```diff
@@ -50,21 +50,20 @@ RUN apt-get -y update && \
...
 RUN useradd --create-home -d /home/sof -u $UID -G sudo sof && \
 echo "sof:test0000" | chpasswd && adduser sof sudo
 ENV HOME /home/sof
 
+# Use ToT alsa utils for the latest topology patches.
+RUN mkdir -p /home/sof/work/alsa && cd /home/sof/work/alsa && \
+git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-lib.git && \
+git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-utils.git && \
+cd /home/sof/work/alsa/alsa-lib && ./gitcompile &&  make install && \
+cd /home/sof/work/alsa/alsa-utils && ./gitcompile &&  make install
```
Comment (xiulipan) [inline scripts/docker_build/sof_builder/Dockerfile:65]:
```suggestion
cd /home/sof/work/alsa/alsa-utils && ./gitcompile &&  make install &&\
chown -R sof:sof /home/sof
```

This works on my machine. Will upload this Docker image

Comment (lgirdwood) [reply scripts/docker_build/sof_builder/Dockerfile:65]: Ok, great - can you fix and upstream this PR too, I wont get a chance until I'm back next week. If so, pls close this one too.

Comment (xiulipan) [reply scripts/docker_build/sof_builder/Dockerfile:65]: @lgirdwood Sure, I also find some build error with GCC 9. So I will need to fix those issues first before upstream the docker to GCC 9

**Comment by xiulipan** [comment]
@lgirdwood Updated this PR as I found I can push to your topic branch.
Blocked by https://github.com/thesofproject/sof/issues/3244

**Comment by lgirdwood** [comment]
@xiulipan are we good to merge ? as #3244 now fixed.

**Comment by xiulipan** [comment]
@lgirdwood Let me build an image and do some test.

**Comment by xiulipan** [comment]
@lgirdwood Good to go, local test pass.

**Pull request merged**